tachyon/zk/plonk/circuit/examples/simple_lookup_circuit_unittest.cc

```diff
@@ -0,0 +1,474 @@
+#include "tachyon/zk/plonk/keys/halo2/pinned_verifying_key.h"
+#include "tachyon/zk/plonk/keys/proving_key.h"
+
+namespace tachyon::zk {
+
+namespace {
+
+constexpr size_t kBits = 3;
+
+class SimpleLookupCircuitTest : public CircuitTest {};
+
+}  // namespace
+
+TEST_F(SimpleLookupCircuitTest, Configure) {
+  ConstraintSystem<F> constraint_system;
+  SimpleLookupConfig<F, kBits> config =
+      SimpleLookupCircuit<F, kBits>::Configure(constraint_system);
+  EXPECT_EQ(config.selector(), Selector::Complex(0));
+  EXPECT_EQ(config.table(), LookupTableColumn(FixedColumnKey(0)));
+  EXPECT_EQ(config.advice(), AdviceColumnKey(0));
+  EXPECT_EQ(constraint_system.num_fixed_columns(), 1);
+  EXPECT_EQ(constraint_system.num_advice_columns(), 1);
+  EXPECT_EQ(constraint_system.num_instance_columns(), 0);
+  EXPECT_EQ(constraint_system.num_selectors(), 1);
+  EXPECT_EQ(constraint_system.num_challenges(), 0);
+  std::vector<Phase> expected_advice_column_phases = {kFirstPhase};
+  EXPECT_EQ(constraint_system.advice_column_phases(),
+            expected_advice_column_phases);
+  EXPECT_TRUE(constraint_system.challenge_phases().empty());
+  EXPECT_TRUE(constraint_system.selector_map().empty());
+  EXPECT_TRUE(constraint_system.gates().empty());
+  std::vector<AdviceQueryData> expected_advice_queries = {
+      AdviceQueryData(Rotation::Cur(), AdviceColumnKey(0)),
+  };
+  EXPECT_EQ(constraint_system.advice_queries(), expected_advice_queries);
+  std::vector<size_t> expected_num_advice_queries = {1};
+  EXPECT_EQ(constraint_system.num_advice_queries(),
+            expected_num_advice_queries);
+  EXPECT_TRUE(constraint_system.instance_queries().empty());
+  std::vector<FixedQueryData> expected_fixed_queries = {
+      FixedQueryData(Rotation::Cur(), FixedColumnKey(0)),
+  };
+  EXPECT_EQ(constraint_system.fixed_queries(), expected_fixed_queries);
+  EXPECT_TRUE(constraint_system.permutation().columns().empty());
+  std::vector<LookupArgument<F>> expected_lookups;
+  {
+    LookupPairs<std::unique_ptr<Expression<F>>> pairs;
+    pairs.emplace_back(
+        ExpressionFactory<F>::Sum(
+            ExpressionFactory<F>::Product(
+                ExpressionFactory<F>::Selector(config.selector()),
+                ExpressionFactory<F>::Advice(
+                    AdviceQuery(0, Rotation::Cur(), config.advice()))),
+            ExpressionFactory<F>::Sum(
+                ExpressionFactory<F>::Constant(F::One()),
+                ExpressionFactory<F>::Negated(
+                    ExpressionFactory<F>::Selector(config.selector())))),
+        ExpressionFactory<F>::Fixed(
+            FixedQuery(0, Rotation::Cur(), config.table().column())));
+    expected_lookups.emplace_back("lookup", std::move(pairs));
+  }
+  EXPECT_EQ(constraint_system.lookups(), expected_lookups);
+  EXPECT_TRUE(constraint_system.general_column_annotations().empty());
+  EXPECT_FALSE(constraint_system.minimum_degree().has_value());
+}
+
+TEST_F(SimpleLookupCircuitTest, Synthesize) {
```

Comment:
While inferring the source of the test data from the circuit's origin is possible, providing a description of the test data would be more helpful to the user. 
It would be appreciated if you could specify which objects' members the data pertains to.

Reply:
Since our structure of `ConstraintSystem`, `ProvingKey` and `VerifyingKey` are not that different, I don't agree that we should put comments for this reason!